### DISCUSSION TITLE
Choosing a segmentation algorithm

### DISCUSSION THREAD

**Discussion by StrawHattM** [etal/cnvkit#1133] [General]
Hello, I was wondering if I could maybe find some help here about how to proceed forward with my analysis.

I have run cnvkit on WGS of 5 tumors, following the recommendations for WGS, against a reference built with a healthy tissue sample. Down the line, I inted to compare the tumors that belong to two different conditions, and figure out phenotype-specific gene-level CNAs. 

When it comes to segmenting, I am encountering the problem of not know which algorithm I should "choose" to keep analyzing from. I have tried using `cbs`, `hmm` and `hmm_tumor`, the latter being the one that from the description in the docs would suit my case better.

In the following graph I have plotted the distribution of log2 values of the segments using the three different algorithms, and that of the .cnr files I fed to the `segment` call. 
![image](https://user-images.githubusercontent.com/109228395/179515727-bdb3c373-b151-4317-a5e8-3dc7ead81c33.png)

These are very aneuploid tumors and that is expected, and from that comparison I think CBS would be the one that reflects the .cnr distribution the best, but I am actually wondering if that is desirable or it's just indicative of more noise than the others. 

Here's the comparison of the number of segments that are detected by each method in each sample: 
![image](https://user-images.githubusercontent.com/109228395/179516237-109fab46-a17a-41b2-b3d6-fa2466f1275b.png)

My questions are: 
 - Is there a particular metric that I could look at in order to determine which segmentation algorithm is optimal?? 
 - Is a segment log2 distribution similar to the bin-level log2 distribution a good quality indication, or does it otherwise mean more noisy samples?
 - Is there any particularity that I should take into account within the analysis to reflect the high aneuploidy levels of my samples?

Thanks a lot for your help!!

**Comment by DineshRavindraRaju** [comment]
@StrawHattM, Have you figured it out? I need to compare tumors from two different conditions, one of which is characterized by aneuploidy

**Comment by StrawHattM** [comment]
Hi @DineshRavindraRaju, sorry it has taken me a while to see this.

So I checked the segment overlap using GRanges, and how many segments would I end up with after removing those which overlapped in more than 25/50/75% of their length.:
![Image](https://github.com/user-attachments/assets/51e9da74-fe56-42de-ab75-deffef2aa00f)

I also checked the number of segments remaining after filtering with this and  -0.5 < log2 < 0.5:

![Image](https://github.com/user-attachments/assets/cc224ebc-2daf-4939-8d1d-69adf408609d)

Out of this, I thought CBS was giving me very normal sample distribution. This could be an advantage if you expect normality, but given my very high aneuploidy tumors I thought this could be an artifact or something the algorithm was forcing. 
Eventually I went with HMM_tumor since HMM was making some weird figures and I decided to trust that HMM tumor was indeed well adapted for well, tumor samples.

What is your experience so far??
